Sources/Markdown/Walker/Walkers/MarkupFormatter.swift

```diff
@@ -492,15 +492,23 @@ public struct MarkupFormatter: MarkupWalker {
 
     /// Get the numeral prefix for a list item if its parent is an ordered list.
     func numeralPrefix(for listItem: ListItem) -> String? {
-        guard listItem.parent is OrderedList else {
+        guard let list = listItem.parent as? OrderedList else {
             return nil
         }
         let numeral: UInt
         switch formattingOptions.orderedListNumerals {
         case let .allSame(n):
-            numeral = n
+            if let origStart = list.start {
+                numeral = UInt(origStart)
+            } else {
+                numeral = n
+            }
```

Comment:
Nit: 

```suggestion
            numeral = list.start.map(UInt.init) ?? n
```

same below